.github/workflows/ci.yml

```diff
@@ -122,34 +125,73 @@ jobs:
 
       - uses: Swatinem/rust-cache@v2
 
-      - name: Prereqs
+      - name: bpf-linker
         run: cargo install bpf-linker --git https://github.com/aya-rs/bpf-linker.git
 
       - uses: taiki-e/install-action@cargo-hack
       - name: Build
         env:
           CARGO_CFG_BPF_TARGET_ARCH: ${{ matrix.arch }}
         run: |
+          set -euxo pipefail
           cargo hack build --package aya-bpf --package aya-log-ebpf \
             --feature-powerset \
             --target ${{ matrix.target }} \
             -Z build-std=core
 
-  integration-test:
-    runs-on: macos-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        # See https://doc.rust-lang.org/cargo/reference/profiles.html for the names
-        # of the builtin profiles. Note that dev builds "debug" targets.
-        profile:
-          - release
-          - dev
+  build-integration-test:
+    runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
 
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly
+          components: rust-src
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: bpf-linker
+        run: cargo install bpf-linker --git https://github.com/aya-rs/bpf-linker.git
+
+      - name: Install dependencies
+        # ubuntu-20.04 comes with clang prior to
```

Comment:
We're on 22.04 now, does this comment still hold?

Reply:
Let's find out! I copied this from a similar stanza in bpf-linker where we're on 20.04. Removed installing clang, let's see what happens.

Reply:
Seems to work.

Reply:
Just kidding, we do need a newer clang still. Added commentary.